resources/asciidoctor/spec/copy_images_spec.rb

```diff
@@ -31,46 +31,115 @@ def copy_attributes(copied)
 
   spec_dir = File.dirname(__FILE__)
 
-  it "copies a file when directly referenced" do
-    copied = []
-    attributes = copy_attributes copied
-    input = <<~ASCIIDOC
-      == Example
-      image::resources/copy_images/example1.png[]
-    ASCIIDOC
-    convert input, attributes,
-        eq("INFO: <stdin>: line 2: copying #{spec_dir}\/resources\/copy_images\/example1.png")
-    expect(copied).to eq([
-        ["resources/copy_images/example1.png", "#{spec_dir}/resources/copy_images/example1.png"],
-    ])
+  shared_context 'convert' do
+    let(:result) do
+      copied = []
+      attributes = copy_attributes copied
+      result = convert_with_logs input, attributes
+      result.push copied
+      result
+    end
+    let(:logs)   { result[1] }
+    let(:copied) { result[2] }
   end
-
-  it "copies a file when it can be found in a sub tree" do
-    copied = []
-    attributes = copy_attributes copied
-    input = <<~ASCIIDOC
-      == Example
-      image::example1.png[]
-    ASCIIDOC
-    convert input, attributes,
-        eq("INFO: <stdin>: line 2: copying #{spec_dir}/resources/copy_images/example1.png")
-    expect(copied).to eq([
-        ["example1.png", "#{spec_dir}/resources/copy_images/example1.png"],
-    ])
+  shared_examples 'copies the image' do
+    include_context 'convert'
+    it 'copies the image' do
+      expect(copied).to eq([
+          [resolved, "#{spec_dir}/resources/copy_images/example1.png"],
+      ])
+    end
+    it 'logs that it copied the image' do
+      expect(logs).to eq("INFO: <stdin>: line #{include_line}: copying #{spec_dir}/resources/copy_images/example1.png")
```

Comment:
The expected result is a bit long, so perhaps it can be put in a `let` block so that it's easier to parse. Also, if the expected result changes, it's easier to locate in the spec file what needs to edited.
For example:

```ruby

let(:expected_copied_image) do
  [[resolved, "#{spec_dir}/resources/copy_images/example1.png"],]
end

it 'logs that it copied the image' do
  expect(logs).to eq(expected_copied_image)
end
```